JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToInteraction/UniqueToInteraction.tsx

```diff
@@ -104,48 +57,73 @@ const UniqueToInteraction = () => {
       [id]: { mode: GridRowModes.View, ignoreModifications: true }
     })
 
-    const editedRow = rows.find(row => row.id === id)
+    const editedRow = rows.find((row: { id: GridRowId }) => row.id === id)
     if (editedRow!.isNew) {
-      setRows(rows.filter(row => row.id !== id))
+      setRows(rows.filter((row: { id: GridRowId }) => row.id !== id))
     }
   }
 
   const processRowUpdate = (newRow: GridRowModel) => {
-    const updatedRow = { ...newRow, isNew: false }
-    setRows(rows.map(row => (row.id === newRow.id ? updatedRow : row)))
+    const { isNew, ...updatedRow } = newRow
+    setRows(
+      rows.map((row: { id: any }) => (row.id === newRow.id ? updatedRow : row))
+    )
     return updatedRow
   }
 
   const handleRowModesModelChange = (newRowModesModel: GridRowModesModel) => {
     setRowModesModel(newRowModesModel)
   }
 
+  const handleRowEditStop: GridEventListener<'rowEditStop'> = (
+    params,
+    event
+  ) => {
+    if (params.reason === GridRowEditStopReasons.rowFocusOut) {
+      event.defaultMuiPrevented = true
+    }
+  }
+
   const columns: GridColDef[] = [
     {
-      field: 'name',
+      field: 'fieldName',
       headerName: 'Name',
       width: 300,
       editable: true,
-      align: 'center',
-      headerAlign: 'center'
+      align: 'left',
+      headerAlign: 'left',
+      valueGetter: params =>
+        params.row.fieldName
+          .replace(/_/g, ' ')
+          .replace(/\b\w/g, (char: string) => char.toUpperCase())
     },
     {
-      field: 'type',
+      field: 'fieldType',
       headerName: 'Type',
       type: 'string',
       width: 300,
       align: 'center',
       headerAlign: 'center',
       editable: false
     },
+
     {
       field: 'index',
       headerName: 'Index',
       type: 'string',
       width: 300,
       align: 'center',
       headerAlign: 'center',
-      editable: false
+      editable: false,
+      valueGetter: params => {
+        const indexGoldenRecord = params.row.indexGoldenRecord
```

Comment:
@NyashaMuusha Can you please refactor and extract this method to a hepler that performs this logic for all index fields in tabs